chain/syncer.go

```diff
@@ -481,3 +431,11 @@ func (syncer *Syncer) exceedsFinalityLimit(chain []types.TipSet) bool {
 	chainHeight, _ := chain[0].Height()
 	return types.NewBlockHeight(chainHeight).LessThan(finalityHeight)
 }
+
+func reverse(tips []types.TipSet) []types.TipSet {
```

Comment:
This exists already in core/policy.go, and without the unnecessary copy. Please extract that one to an exported function in `chain` somewhere and use it instead.